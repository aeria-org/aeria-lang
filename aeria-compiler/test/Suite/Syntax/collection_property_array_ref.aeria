collection CollectionPropertyArrayRefTest2 {
  properties {
    name str
  }
}

collection CollectionPropertyArrayRefTest {
  properties {
    prop []CollectionPropertyArrayRefTest2
  }
}
