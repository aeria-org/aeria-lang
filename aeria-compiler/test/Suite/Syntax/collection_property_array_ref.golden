{
  "kind": "Program",
  "collections": [
    {
      "properties": [
        {
          "type": {
            "kind": "PString"
          },
          "name": {
            "value": "name",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": []
        }
      ],
      "name": {
        "value": "CollectionPropertyArrayRefTest2",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    },
    {
      "properties": [
        {
          "type": {
            "type": {
              "ref": {
                "value": "CollectionPropertyArrayRefTest2",
                "kind": "CollectionName"
              },
              "kind": "PRef"
            },
            "kind": "PArray"
          },
          "name": {
            "value": "prop",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": []
        }
      ],
      "name": {
        "value": "CollectionPropertyArrayRefTest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
